Add <input type=checkbox switch> macOS focus ring support

annevk opened the pull request:
#### 8cbb6b66a695b63eb9a16a9f955d37c9c391e176
<pre>
Add &lt;input type=checkbox switch&gt; macOS focus ring support
<a href="https://bugs.webkit.org/show_bug.cgi?id=266139">https://bugs.webkit.org/show_bug.cgi?id=266139</a>

Reviewed by Aditya Keerthi.

I discovered that the approach taken in 269869@main to teach
RenderTheme is[State]() checks about switches was wrong. Rather than
changing individual state checks, it should have modified
extractControlStyleForRenderer() instead. By correcting that here the
correct thing will happen in RenderTheme for isFocused().

As a drive-by fix this removes RenderThemeMac::updateFocusedState()
which does not appear to be used and was confusing me.

* Source/WebCore/platform/graphics/mac/controls/SwitchTrackMac.mm:
(WebCore::SwitchTrackMac::draw):
* Source/WebCore/rendering/RenderTheme.cpp:
(WebCore::RenderTheme::extractControlStyleForRenderer const):
(WebCore::RenderTheme::isEnabled const):
(WebCore::RenderTheme::isFocused const):
(WebCore::RenderTheme::isPressed const):
* Source/WebCore/rendering/RenderThemeMac.h:
* Source/WebCore/rendering/RenderThemeMac.mm:
(WebCore::RenderThemeMac::updateFocusedState): Deleted.

Canonical link: <a href="https://commits.webkit.org/271884@main">https://commits.webkit.org/271884@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/e6cf43288943659b1c5b01b9b3ebccb8cc03c370

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/29936 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/8602 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/31254 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/32448 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/27071 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/30596 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/10788 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/5857 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/27121 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/30230 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/7210 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/25543 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/6152 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/6319 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/26632 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/33784 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/27305 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/27066 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/32494 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/6253 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/50/builds/4430 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [❌ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/30284 "Found 1 new API test failure: /WebKitGTK/TestWebKitAccessibility:/webkit/WebKitAccessibility/accessible/event-listener (failure)") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/7989 "Built successfully") | | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/7091 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/6995 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/6779 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

pxlcoder commented on Source/WebCore/platform/graphics/mac/controls/SwitchTrackMac.mm:
```diff
@@ -190,6 +192,12 @@
     context.drawConsumingImageBuffer(WTFMove(trackImage), inflatedTrackRect.location());
+
+    if (isFocused) {
+        // FIXME: This does not incorporate the style data as ControlMac::drawCellFocusRing() does.
```
I don't think this `FIXME` is needed? I'm not aware of any style data we would incorporate here, since the things in `drawCellFocusRing` are specific to `NSCell` configuration.

annevk replied:
Okay, I can remove it. I wasn't sure if there were any settings we would not be taking into consideration this way.

annevk committed:
Add <input type=checkbox switch> macOS focus ring support
https://bugs.webkit.org/show_bug.cgi?id=266139

Reviewed by Aditya Keerthi.

I discovered that the approach taken in 269869@main to teach
RenderTheme is[State]() checks about switches was wrong. Rather than
changing individual state checks, it should have modified
extractControlStyleForRenderer() instead. By correcting that here the
correct thing will happen in RenderTheme for isFocused().

As a drive-by fix this removes RenderThemeMac::updateFocusedState()
which does not appear to be used and was confusing me.

* Source/WebCore/platform/graphics/mac/controls/SwitchTrackMac.mm:
(WebCore::SwitchTrackMac::draw):
* Source/WebCore/rendering/RenderTheme.cpp:
(WebCore::RenderTheme::extractControlStyleForRenderer const):
(WebCore::RenderTheme::isEnabled const):
(WebCore::RenderTheme::isFocused const):
(WebCore::RenderTheme::isPressed const):
* Source/WebCore/rendering/RenderThemeMac.h:
* Source/WebCore/rendering/RenderThemeMac.mm:
(WebCore::RenderThemeMac::updateFocusedState): Deleted.

Canonical link: https://commits.webkit.org/271884@main
